shell/platform/windows/game_pad_cursor_winuwp.h

```diff
@@ -0,0 +1,130 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_GAME_PAD_CURSOR_WINUWP_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_GAME_PAD_CURSOR_WINUWP_H_
+
+#include <DispatcherQueue.h>
+#include <winrt/Windows.Graphics.Display.h>
+#include <winrt/Windows.System.Profile.h>
+#include <winrt/Windows.UI.Composition.h>
+#include <winrt/Windows.UI.Core.h>
+#include <winrt/Windows.UI.ViewManagement.Core.h>
+#include "winrt/Windows.System.Threading.h"
+
+#include "flutter/shell/platform/windows/display_helper_winuwp.h"
+#include "flutter/shell/platform/windows/flutter_windows_view.h"
+#include "flutter/shell/platform/windows/game_pad_winuwp.h"
+
+namespace flutter {
+
+// A class abstraction for a mouse cursor driven by a gamepad.  This abstraction
+// is primarily intended for devices like XBOX that do not have traditional
+// mouse input.
+class GamepadCursorWinUWP {
+ public:
+  GamepadCursorWinUWP(
+      WindowBindingHandlerDelegate* view,
+      DisplayHelperWinUWP* displayhelper,
+      winrt::Windows::UI::Core::CoreWindow const& window,
+      winrt::Windows::UI::Composition::Compositor const& compositor,
+      winrt::Windows::UI::Composition::VisualCollection const& rootcollection);
+
+ private:
+  // Pointer to a DisplayHelperWinUWP object used to get windowbounds, dpi and
+  // other display related aspects.
+  DisplayHelperWinUWP* display_helper_;
+
+  // Current active compositor. nullptr if not set.
+  winrt::Windows::UI::Composition::Compositor compositor_{nullptr};
+
+  // Window that is hosting this emulated cursor.
+  winrt::Windows::UI::Core::CoreWindow window_{nullptr};
+
+  // The root of the composition tree that this class will add a custor visual
+  // to.
+  winrt::Windows::UI::Composition::VisualCollection root_collection_{nullptr};
+
+  // Called when the user is interacting with the GamePad and moving the mouse
+  // cursor.  Prevents the mouse cursor from being hidden by resetting the
+  // inactivity timer.
+  void SetCursorTimeout();
```

Comment:
Ultimately, we will want more control eg turn mouse emulation on and off, off for the case where the flutter app is consuming the gamecontroller input directly rather than us emulating a mouse and injecting mouse events.  ATM this change is an MVP to stand up basic xbox support, would prefer to not grow the scope in this PR particularly as that would require framework integration.